crates/astria-account-monitor/src/account_monitor.rs

```diff
@@ -267,27 +259,27 @@ impl<T: Display> Display for QueryResponse<T> {
     }
 }
 
-// Implement Display for AccountInfo
+#[derive(Debug, Clone)]
+struct AccountInfo {
+    nonce: QueryResponse<u32>,
+    balance: QueryResponse<Vec<AssetBalance>>,
+}
+
 impl Display for AccountInfo {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "nonce = {}, balance = {}", self.nonce, self.balance)
+        write!(f, "nonce = {}, balance = {:?}", self.nonce, self.balance)
     }
 }
 
 async fn get_latest_balance(
```

Comment:
Bit confusing naming because this is getting multiple balance**s**, not just a single balance.